src/app/task.service.ts

```diff
@@ -72,24 +72,19 @@ export class TaskService {
       });
   }
 
-  private loadTasksFromAPI() {
-    this.http.get<TaskList>("http://localhost:3001/allTasks/")
-      .pipe(
-        catchError((error) => {
-          console.error('Error fetching tasks from API:', error);
-          return [];
-        })
-      )
-      .subscribe({
-        next: (response: TaskList) => {
-          this.taskList = response.tasks;
-        },
-        error: (error) => {
-          console.log(error);
-        },
-        complete: () => {
-          console.log('Task loading complete');
-        }
-      });
+  async loadTasksFromAPI() {
```

Comment:
you can remove the async keyword here and await just return the this.http.get<...>(...)

it will return an Observable, you can subscribe to this observable in the component like this:

this.yourService.loadTasksFromAPI().pipe(take(1)).subscribe();

this way it will only subscribe until the first value is taken and then automatically unsubscribe.

after the subscription it will update the taskListSubject which will update the Observable taskList$ in your service (and component) then you just use that Observable in the component template to render data 😃